Auto-batching - Enable feature by default and remove unwanted options (#162)
meilisearch/specifications: text/0096-auto-batching.md

#### 3.1.2. Impacts on `task` API resource

> - The different tasks grouped in a batch are processed within the same transaction. But if a task fails within a batch, the whole batch does not fail, only the related task.

Author: @Kerollmops if there is any new type of error created (we would need to add [here in the docs](https://docs.meilisearch.com/reference/api/error_codes.html) and [here in the spec](https://github.com/meilisearch/specifications/blob/main/text/0061-error-format-and-definitions.md)), can you let me know, because I haven't updated the spec accordingly since I don't know what changed regarding the errors?

Reviewer: It doesn't seem that any error code has been changed or added.
Below is the message returned by the failed task processed in a partially valid batch:
```json
"error": {
        "message": "Document doesn't have a `id` attribute: `{\"title\":\"lol\",\"author\":\"guy\",\"new_field\":\"bruh\"}`.",
        "code": "missing_document_id",
        "type": "invalid_request",
        "link": "https://docs.meilisearch.com/errors#missing_document_id"
}
```

- Extends it for all consecutive payload types.
- Add a filter capability by `batchUid` on the `/tasks` endpoints.

> - <s>Do not fail the entire transaction if a document is not valid. Report the documents that could not be indexed to the user.</s>

Author: Don't know if this format is ok for you @gmourier to say "it's done" without completely remove them

Reviewer: The current behavior fails the entire task if one document is invalid. However, an entire batch doesn't fail anymore if a contained task failed.

Reviewer 2: @curquiza This format might suggest that we have decided not to implement future possibilities. WDYT?

> - <s>Do not fail the entire transaction if a document is not valid. Report the documents that could not be indexed to the user.</s>

Reviewer 2: ```suggestion
- Report documents that could not be indexed to the user in a more precise manner.
```

> - <s>Enable auto-batching by default.</s>

Reviewer 2: ```suggestion
```
Ps: Integrate this suggestion given our final decision regarding the discussion in the previous comment about the format
